notebook/templates/edit.html

```diff
@@ -17,10 +19,16 @@
 {% endblock %}
 
 {% block headercontainer %}
-
 <span id="save_widget" class="pull-left save_widget">
-    <span class="filename"></span>
-    <span class="last_modified"></span>
+    <span id="notebook_name" class="filename"></span>
+    <span class="checkpoint_status"></span>
+    <span class="autosave_status"></span>
+</span>
+
+<span id="kernel_logo_widget">
+  {% block kernel_logo_widget %}
+  <img class="current_kernel_logo" src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"/>
+  {% endblock %}
```

Comment:
The changes here to the save widget shouldn't be needed, since they are all notebook-specific.